containers/ecr-viewer/src/app/api/migrate-db/migrations/core/20250430160000_condition_fk.ts

```diff
@@ -0,0 +1,86 @@
+import { Kysely, sql } from "kysely";
+
+import { AnyDb } from "@/app/data/metadataDb/database";
+import { dbNamespace, dbDialect } from "@/app/data/metadataDb/utils/db-config";
+import { getTable } from "@/app/data/metadataDb/utils/db-metadata";
+
+/**
+ * Add condition_code column and foreign key constraint to ecr_rr_conditions.
+ * @param db - the database connection
+ */
+export async function up(db: Kysely<AnyDb>): Promise<void> {
+  const schema = dbNamespace();
+  const table = await getTable(db, dbNamespace(), "ecr_rr_conditions");
+
+  const foreignKeyCheck =
+    !!table && table.columns.some((c) => c.name === "condition_code");
+
+  if (foreignKeyCheck) return;
+
+  const _db = db.withSchema(schema);
+
+  await _db.schema
+    .alterTable("ecr_rr_conditions")
+    .addColumn("condition_code", "varchar(20)", (cb) =>
+      cb.references("condition_reference.code"),
+    )
+    .execute();
+
+  if (dbDialect() === "postgres") {
+    await sql`
+      CREATE OR REPLACE FUNCTION set_condition_code()
+      RETURNS TRIGGER AS $$
+      BEGIN
+        IF NEW.condition_code IS NULL THEN
+          SELECT code INTO NEW.condition_code
+          FROM ${sql.raw(schema)}.condition_reference
+          WHERE condition_reference.condition_name = NEW.condition
+          LIMIT 1;
+        END IF;
+        RETURN NEW;
+      END;
+      $$ LANGUAGE plpgsql;
+    `.execute(_db);
+
+    await sql`
+      CREATE TRIGGER trg_set_condition_code
+      BEFORE INSERT ON ${sql.raw(schema)}.ecr_rr_conditions
```

Comment:
instead of doing this with triggers can we just do an update on the existing table and instead update the metadata parser to just pull out the code from the fhir bundle so we get the original source of truth from what was there and then save that as we do regular data for go forward?